Interchange 1.0 roadmap (#515)
openforcefield/openforcefield.org: content/community/news/general/interchange-1.0-roadmap.md

---

> date: "2025-02-27T00:00:00+00:00"

Reviewer: (not blocking) Could be good to update this before merging, since it's already decently far in the past.

Author: 45a4b0dd10be3d93c0dfdb87cd87b9cb62af390f

title: "Interchange 1.0 roadmap"
tags: ["news","general","newsletter","Open Force Field Initiative"]
categories: ["news"]
draft: false
description: "The roadmap for Interchange version 1.0"
weight: 10
author: "Matt Thompson"
thumb: "openff-rocket.png"
---

Interchange is a project (and object) for storing, manipulating, and converting molecular mechanics data and currently functions as the interoperability backend within OpenFF's software stack. It has been used for years as the workhorse behind `ForceField.create_openmm_system` but has also been used for preparing GROMACS, Amber, and LAMMPS simulations in both academic research groups and industry. Recently, some users have chained together `Interchange.from_openmm` and `Interchange.combine` in workflows to prepare systems with both OpenFF and external tooling which mix SMIRNOFF and other force fields.

> Interchange has been in various states of alpha/beta testing and pre-release for years, but it has quietly performed well since it was fully integrated in the OpenFF software stack in 2023. Its API has become more and more stable since then major features have been implemented. The time has come for a 1.0 release which better communicates that core features of Interchange are ready for general use. This is far from the end of the project - interoperability in this space will always be an ongoing effort - but it is a good time to solidify the API and feature scope of a first major release.

Reviewer: ```suggestion
Interchange has been in various states of alpha/beta testing and pre-release for years, but it has quietly performed well since it was fully integrated in the OpenFF software stack in 2023. Its API has become increasingly stable since then, and major features have been implemented. The time has come for a 1.0 release which better communicates that core features of Interchange are ready for general use. This is far from the end of the project - interoperability in this space will always be an ongoing effort - but it is a good time to solidify the API and feature scope of a first major release.
```

In February, the OpenFF Infrastructure team conducted a survey directed at users of OpenFF software and force fields to better understand what features should go into the first stable 1.0 release of Interchange. We were happy to find that people were broadly satisfied with

- the scope of features provided by Interchange
- the ease of installation
- the quantity and quality of provided examples and documentation
- generally how well it enables the use of OpenFF force fields in different molecular simulation engines.

> We did not find major gaps in feature coverage that necessitate major feature development, but we did take some helpful suggestions for minor feature additions which are included in the roadmap below. This roadmap also includes committment to a set of core features that we believe compose most use cases. Some other featuers which are used by comparatively fewer people are left in place. Nothing significant is slated for removal with 1.0.

Reviewer: ```suggestion
We did not find major gaps in feature coverage that necessitate major feature development, but we did take some helpful suggestions for minor feature additions which are included in the roadmap below. This roadmap also includes commitment to a set of core features that we believe compose most use cases. Some other features which are used by comparatively fewer people are left in place. Nothing significant is slated for removal with 1.0.
```

Minor features

- Virtual sites processed when calling `Interchange.from_openmm`
- Packmol wrapper functions part of a public API

> - Improved runtime performance on system containing biopolymers

Reviewer: ```suggestion
- Improved runtime performance on systems containing biopolymers
```

- Improved interoperability testing on systems containing biopolymers

Maintenance committments

> - Complete (current) SMIRNOFF specification

Reviewer: (not blocking) The current SMIRNOFF spec includes custom functional forms for eg. torsions, which we don't currently support. Is there a way to reword this to better capture what is meant here?

Author: Tried to pin this down in 0b87fb8bc164b1b7f556e20fa6d136c8b32b6f7b
